third_party/tlx/tutorials/hopper-fa-ws-pipelined-pingpong_test.py

```diff
@@ -174,8 +176,22 @@ def _attn_fwd_ws_pipelined_pingpong(sm_scale, M,  #
 
                 # compute qk for the current iteration
                 k_tile = tlx.local_trans(k_tiles[k_buf_id])
+
+                if cid == 0:
+                    # Consumer 0 waits for Consumer 1 to reach synchronization point at barrier 9.
+                    tlx.named_barrier_wait(9, 256)
+                else:
+                    # Then waits at barrier 10 until Consumer 0 finishes issuing its async_dot.
+                    tlx.named_barrier_wait(10, 256)
+
                 qk = tlx.async_dot(q_tiles[cid], k_tile)
 
+                if cid == 0:
+                    # After issuing async_dot, Consumer 0 signals barrier 10 to unblock Consumer 1.
+                    tlx.named_barrier_arrive(10, 256)
+                else:
+                    tlx.named_barrier_arrive(9, 256)
+
```

Comment:
For acc, adding pingpong does not lead to strict performance improvement, so I left them out.
The pass #668 enforces pingpong also for acc, though.

Reply:
I see. Perhaps the pingpong here can already interview the two consumers.